backend/routers/users.js

```diff
@@ -0,0 +1,144 @@
+const { Router } = require('express');
+const jwt = require('jsonwebtoken');
+const bcrypt = require('bcryptjs');
+const createError = require('http-errors');
+const { UserModel } = require('../models');
+const { auth } = require('../middlewares');
+
+const router = Router();
+
+async function getUser(req, res, next) {
+  const id = Number.parseInt(req.params.uid, 10);
+  try {
+    if (!id) {
+      throw createError(400, 'Please specify user id');
+    }
+    const user = await UserModel.find({ id });
+    if (user.length === 0) {
+      throw createError(404, 'User not found with that id');
+    }
+    res.send(user[0]);
+  } catch (error) {
+    next(error);
+  }
+}
+
+function isValidEmail(email) {
+  const emailChecker = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+  return emailChecker.test(String(email).toLowerCase());
+}
+
+async function register(req, res, next) {
+  const {
+    username, password, email, kingdomName,
+  } = req.body;
+  try {
+    if (!username && !password) {
+      throw createError(400, 'Username and password are required.');
+    }
+    if (!username) {
+      throw createError(400, 'Username is required.');
+    }
+    if (!password) {
+      throw createError(400, 'Password is required.');
+    }
+    if (!email) {
+      throw createError(400, 'Email is required.');
+    }
+    if (!isValidEmail(email)) {
+      throw createError(400, 'Email is not valid.');
+    }
+    if (password.length < 8) {
+      throw createError(400, 'Password must be at least 8 characters.');
+    }
+    if (await UserModel.findOne({ username })) {
+      throw createError(400, 'Username is already taken.');
+    }
+    if (await UserModel.findOne({ email })) {
+      throw createError(400, 'Email is already taken.');
+    }
+    const passwordHash = await bcrypt.hash(password, 8);
+    const user = await UserModel.create({
+      email,
+      username,
+      password: passwordHash,
+      kingdomName: kingdomName || `${username}'s kingdom`,
+      kingdomList: [],
+    });
+    user.password = undefined;
```

Comment:
```suggestion
    delete user.password;
```

instead of replacing the value to `undefined`, delete it from the user object, so it won't be in the token at all